google/cloud/storage/blob.py

```diff
@@ -804,7 +804,7 @@ def _extract_headers_from_download(self, response):
 
         digests = {}
         for encoded_digest in x_goog_hash.split(","):
-            match = re.match(r"(crc32c|md5)=([\w\d]+)==", encoded_digest)
+            match = re.match(r"(crc32c|md5)=([\w\d]+==)", encoded_digest)
```

Comment:
Could you clarify what the issue was in this case?

Reply:
Hmmm, the [API docs for the `Object` representation](https://cloud.google.com/storage/docs/json_api/v1/objects#crc32c) state:
```
CRC32c checksum, as described in RFC 4960, Appendix B; encoded using base64 in big-endian byte order. For more information about using the CRC32c checksum, see Hashes and ETags: Best Practices.
```

[Likewise for `md5hash`](https://cloud.google.com/storage/docs/json_api/v1/objects#md5hash):
```
MD5 hash of the data; encoded using base64. For more information about using the MD5 hash, see Hashes and ETags: Best Practices.
```

The `Blob.crc32c` and `Blob.md5_hash` properties don't do any decoding / remapping of the `base64`-encoded strings (which is maybe a pain for users), but leaving the `==` padding characters in place looks consistent with how we normally store those values.

Reply:
@frankyn `match = re.match(r"(crc32c|md5)=([\w\d]+)==", encoded_digest)` method removes the `==` padding from the end so when user tries to update the file content after download it throws `google.api_core.exceptions.BadRequest: 400` error because the crc32 and md5 's hash value mismatched so i changed regex which doesn't remove  the padding, then  user can upload or update content of blob multiple times.